### PR TITLE
Building better ML models : Power of Ensembles - Bargava

### PR DESCRIPTION
**Pull request by rouseguy**
My article : Building better machine learning models: Power of ensembles for Poland PyCon 2015. This goes along with the talk that I am giving.

### CODE REVIEW

File: 2015/presentations/ensemble_models.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+![alt text](ensemble_1.png)
+
+Multiple model outputs are combined to create the final prediction. The individual(single) models are called base models. How the various base models are combined(ensembled) to create the final prediction determines the quality/accuracy of ensemble learning.
+
+The key ingredient to ensemble is to select the base models with significant diversity in their predictions. If the individual base learners are similar, they add little information. Worse, they tend to reinforce the same error present within them.
+
+###Motivation Example
+Before going into the various ways of ensembling, let's take a step back and realize that ensembles do work with a simple example.
+
+Let's consider an example of classifying a credit card transaction as fradulent or not. The prediction has to be made on 10 transactions. Let's take the case where we already know those 10 cases to be fradulent. Encoding 1 as fradulent and 0 as not-fraudulent, let there be three base model ouputs, one each from Logistic Regression, Random Forest and Gradient Boosting, each having 70% accuracy.
```
Comment (pigletto) [inline 2015/presentations/ensemble_models.md]: "fradulent" -> "fraudulent"  (x3)

File: 2015/presentations/ensemble_models.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+| 1           | 1       |      0            | 1  |
+| 1           | 1       |      1            | 1  |
+| 0           | 0       |      1            | 0  |
+| 1           | 1       |      0            | 1  |
+| 1           | 1       |      1            | 1  |
+| 1           | 0       |      1            | 1  |
+| 0           | 1       |      1            | 1  |
+
+
+A key point to be stressed again is that model diversity is important. If the models were similar, the output would've been similar and there would've been no improvement i the output.
```
Comment (pigletto) [inline 2015/presentations/ensemble_models.md]: "i" -> "in"

File: 2015/presentations/ensemble_models.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+### Different ways to create ensemble models
+
+
+####Base Models
+
+Let's now discuss some of the ways to create the base models, to ensure that they have enough diversity
+
+1. **Different training sets**: If there's enough data, the same model can be run on different training examples. In practice, transformation of original training set is carried out to create multiple training sets. (*Widely used*)
+2. **Different algorithms/algorithm diversity**: Choose diverse algorithms. `LinearRegression`, `LogisticRegression` are diverse from `DecisionTree`, `RandomForest` etc (Regression Vs Tree-based). `SVM` is another class of algorithms. Also, `Regression` could be L1 or L2. The basis function for `SVM` could be radial or linear.  (*Widely used*)
+3. **Different parameter setups**: Called the hyperparameters, the parameters of the model can be varied to obtained different model outputs. An example would be to build `RandomForest` with shallow trees versus building it with denser trees (less trees vs more trees) (*Widely used*)
```
Comment (pigletto) [inline 2015/presentations/ensemble_models.md]: "obtained" -> "obtain"

File: 2015/presentations/ensemble_models.md (hunk 1 above)
Comment (soltysh) [inline 2015/presentations/ensemble_models.md]: s/not-fraudulent/non-fraudulent

File: 2015/presentations/ensemble_models.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+4. **Algorithm randomization**: A lot of the models have randomness playing a key part and the model can be run with different random initialization. (*Not so widely used*)
+5. **Feature Sampling** Using subset of columns for different model runs produce different model outputs and can help in estimating and containing variance of model predictions(*Widely used*)
```
Comment (soltysh) [inline 2015/presentations/ensemble_models.md]: model runs, produce...

**Comment by soltysh** [comment]
Some small typos but generally LGTM.

**Comment by soltysh** [comment]
Thanks Bargava!

**Comment by rouseguy** [comment]
I should've proof-read better. Thanks for all the corrections.

**Pull request merged**